### PR TITLE
Slicer: remove classification grid search, simplify metadata, add ana…

### PR DESCRIPTION
**Pull request by luca-rossi**
…lytics

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CODE_OF_CONDUCT.md) document.
- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/Giskard-AI/ai-inspector/blob/master/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

**Comment by mattbit** [comment]
Values of Gini impurity reported by @luca-rossi by experimenting on 25 different models/datasets:

```
Min: 0.13991769547325106, Max: 0.6429414741103052, Mode: %s Classification
Min: 0.42965154503616043, Max: 0.6427296171394565, Mode: %s Classification
Min: 0.19451530612244894, Max: 0.602788279773157, Mode: %s Classification
Min: 0.19132653061224492, Max: 0.6244444444444445, Mode: %s Classification
Min: 0.30583900226757366, Max: 0.6261111111111111, Mode: %s Classification
Min: 0.24058810425484523, Max: 0.6131656804733727, Mode: %s Classification
Min: 0.5181402848696587, Max: 0.6404081632653061, Mode: %s Classification
Min: 0.08317580340264652, Max: 0.6397334443981675, Mode: %s Classification
Min: 0.08317580340264652, Max: 0.6294444444444445, Mode: %s Classification
Min: 0.25566106647187725, Max: 0.6416831032215647, Mode: %s Classification
Min: 0.11072664359861595, Max: 0.6127777777777778, Mode: %s Classification
Min: 0.09280190362879237, Max: 0.5509259259259259, Mode: %s Classification
Min: 0.3835616438356164, Max: 0.6309233229526134, Mode: %s Classification
Min: 0.353185595567867, Max: 0.5996430696014277, Mode: %s Classification
Min: 0.27055150884495316, Max: 0.6038062283737025, Mode: %s Classification
Min: 0.4377777777777778, Max: 0.5553633217993079, Mode: %s Classification
Min: 0.4881299458558934, Max: 0.6643598615916955, Mode: %s Classification
Min: 0.5639500297441999, Max: 0.6663648709823449, Mode: %s Classification
Min: 0.05551020408163265, Max: 0.6137543252595156, Mode: %s Classification
Min: 0.29714285714285715, Max: 0.5887573964497042, Mode: %s Classification
Min: 0.49020408163265305, Max: 0.66219723183391, Mode: %s Classification
Min: 0.060546875, Max: 0.6385041551246537, Mode: %s Classification
Min: 0.5289256198347108, Max: 0.6659729448491155, Mode: %s Classification
Min: 0.3949652777777778, Max: 0.5, Mode: %s Classification
Min: 0.36815193571950333, Max: 0.5, Mode: %s Classification
Min: 0.43827611395178967, Max: 0.5, Mode: %s Classification
Min: 0.4054820415879017, Max: 0.49991566874683757, Mode: %s Classification
Min: 0.0, Max: 0.12444444444444447, Mode: %s Classification
Min: 0.13991769547325106, Max: 0.6377777777777778, Mode: %s Classification
Min: 0.1721536351165981, Max: 0.6116071428571428, Mode: %s Classification
Min: 0.257201646090535, Max: 0.65087890625, Mode: %s Classification
Min: 0.13991769547325106, Max: 0.6385041551246537, Mode: %s Classification
Min: 0.12192816635160686, Max: 0.6259040105193951, Mode: %s Classification
Min: 0.1472, Max: 0.6029877772747849, Mode: %s Classification
Min: 0.18000000000000005, Max: 0.6247165532879819, Mode: %s Classification
Min: 0.3077500469131169, Max: 0.625, Mode: %s Classification
Min: 0.45369406867845996, Max: 0.6660494015580857, Mode: %s Classification
Min: 0.5573169789189902, Max: 0.6458768123557999, Mode: %s Classification
Min: 0.1022644265887509, Max: 0.5, Mode: %s Classification
Min: 0.14201183431952658, Max: 0.4921875, Mode: %s Classification
Min: 0.32498740236835477, Max: 0.6145404663923182, Mode: %s Classification
Min: 0.3673094582185491, Max: 0.5, Mode: %s Classification
Min: 0.3161865569272977, Max: 0.6432291666666667, Mode: %s Classification
Min: 0.22646604938271608, Max: 0.5707649876614893, Mode: %s Classification
```

File: giskard/slicing/tree_slicer.py [hunk 1]
```diff
@@ -79,24 +80,34 @@ def find_slices(self, features, target=None):
...
-        dt = gs.best_estimator_
+            dt.fit(data.loc[:, features], data.loc[:, target])
 
         # Need at least a split, otherwise return now.
         if dt.tree_.node_count < 2:
             logger.debug("No split found, stopping now.")
             return []
 
+        analytics.track(
+            "scan:dt_params",
```
Comment (mattbit) [inline giskard/slicing/tree_slicer.py:102]: let's make this more explicit or we won't know what this is

File: giskard/datasets/metadata/text_metadata_provider.py [hunk 2]
```diff
@@ -19,10 +18,6 @@ def generate_metadata(self, values: pd.Series):
         return pd.DataFrame(
             {
                 "text_length": values.map(len),
-                "avg_word_length": values.map(_avg_word_length),
-                "charset": pd.Categorical(values.map(_detect_charset)),
-                "avg_whitespace": values.map(_avg_whitespace),
-                "avg_digits": values.map(_avg_digits),
```
Comment (mattbit) [inline giskard/datasets/metadata/text_metadata_provider.py:25]: Let's keep this for now, I would open a separate PR. I would also keep the charset metadata initially.

**Review by mattbit — CHANGES_REQUESTED** [review]

File: giskard/slicing/tree_slicer.py (hunk 1 above)
Comment (luca-rossi) [reply giskard/slicing/tree_slicer.py:102]: Maybe we should remove it, we don't need it anymore

File: giskard/slicing/tree_slicer.py [hunk 3]
```diff
@@ -79,24 +81,39 @@
...
+            dt.fit(data.loc[:, features], data.loc[:, target])
+            mode = "Classification"
 
         # Need at least a split, otherwise return now.
         if dt.tree_.node_count < 2:
             logger.debug("No split found, stopping now.")
             return []
 
+        # track analytics with a 10% probability
+        if np.random.rand() < 0.1:
```
[the remark this answers is not in the document]
Comment (mattbit) [reply giskard/slicing/tree_slicer.py:105]: they are right: get a generator with `np.random.default_rng()`

File: giskard/slicing/tree_slicer.py [hunk 4]
```diff
@@ -79,24 +81,39 @@ def find_slices(self, features, target=None):
...
+            analytics.track(
+                "scan:tree_slicer_params",
+                {
+                    "n_samples": len(data),
+                    "min_samples": min_samples,
+                    "node_count": dt.tree_.node_count,
+                    "impurity": dt.tree_.impurity.tolist(),
```
Comment (mattbit) [inline giskard/slicing/tree_slicer.py:112]: is impurity always set?

File: giskard/slicing/tree_slicer.py [hunk 5]
```diff
@@ -59,6 +60,7 @@ def find_slices(self, features, target=None):
         if len(data) < min_samples:
             return []
 
+        mode = None
```
Comment (mattbit) [inline giskard/slicing/tree_slicer.py:63]: this is ugly, just pass the class name to the telemetry

**Review by mattbit — CHANGES_REQUESTED** [review]

File: giskard/slicing/tree_slicer.py (hunk 4 above)
Comment (luca-rossi) [reply giskard/slicing/tree_slicer.py:112]: Yes

File: giskard/slicing/tree_slicer.py (hunk 5 above)
Comment (luca-rossi) [reply giskard/slicing/tree_slicer.py:63]: But that wouldn't tell us if it's classification or regression and I think that's important to know (I agree it's ugly though)

**Review by mattbit — APPROVED** [review]

File: giskard/slicing/tree_slicer.py (hunk 5 above)
Comment (mattbit) [reply giskard/slicing/tree_slicer.py]: Fixed

**Pull request merged**